### DISCUSSION TITLE
Offline installation of Moose compilation error

### DISCUSSION THREAD

**Discussion by 09liu** [idaholab/moose#25391] [Q&A Getting Started]
I install moose on a computer without an internet connection and successfully compile petsc and libmesh([https://mooseframework.inl.gov/getting_started/installation/offline_installation.html](url)). But I meet an error when I compile moose. What caused this error and how can I fix it. @hugary1995 
[output.txt](https://github.com/idaholab/moose/files/12506181/output.txt)

**Comment by lindsayad** [comment]
This looks to me like a mismatch between your MOOSE version and your libMesh/TIMPI version. Did you recursively update submodules?

**Reply by 09liu** [reply]
I meet the same error as [https://github.com/idaholab/moose/discussions/21342?sort=top](url). So I install libtirpc-1.1.4-5.el8.x86_64 and libtirpc-devel-1.1.4-5.el8.x86_64 before compile libmesh. In addition, I install mpich-3.4.2 due to my computer's system is RHEL8.4 without mpich. Is it because of the problem with their version？

**Reply by GiudGiud** [reply]
Alex is talking about MOOSE and libmesh versions.
- How did you obtain both codes to install offline?
- Can you give us the current commit for moose and libmesh? `cd` into each folder and type `git log HEAD`

**Reply by lindsayad** [reply]
Also will need to know the TIMPI hash. You can run

git submodule status —recursive

**Reply by 09liu** [reply]
I want to use the program developed on the updated moose in July 2022, so I downloaded the latest version of moose first.
[https://mooseframework.inl.gov/getting_started/installation/offline_installation.html](url)
`git clone https://github.com/idaholab/moose.git
cd moose
git checkout master
git submodule update --init
git submodule foreach --recursive git submodule update --init`
Then I use the following command for version rollback.
`git reset --hard ccf25ce`
Finally I install moose  on a computer without an internet connection. The following are the results I obtaine using the command.

1. moose: 
commit ccf25cee294df35a2796b76e2df27558dac45122 (HEAD -> master)
Merge: 27d88630d4 3bb5876c5c
Author: moosetest <bounces@inl.gov>
Date:   Wed Jul 13 22:46:33 2022 -0600

    Merge commit '3bb5876c5c3f29649f426ff5535f8a3007658acb'
2. libmesh: 
commit fd900b54a02b9d1b99f0a4371d4acb4761e57b0f (HEAD)
Merge: c965dd619 9341cc9bc
Author: Roy H. Stogner <Roy.Stogner@inl.gov>
Date:   Tue Jul 25 08:06:06 2023 -0500

    Merge pull request #3604 from roystgnr/l2_hierarchic_prism
    
    Hierarchic (and L2_HIERARCHIC) FE prism support
3. git submodule status: 
[liu@localhost moose_22.7]$ git submodule status
+08357135db54e430770ee8433f5e6e9735e2c931 large_media (remotes/origin/phase_field_old_wiki-128-g0835713)
+fd900b54a02b9d1b99f0a4371d4acb4761e57b0f libmesh (cpp03_final-8761-gfd900b54a)
 f855b95493736b087b8ccc16dc6c5b29bc4b5aa8 petsc (v3.16.5-15-gf855b954937)

**Reply by GiudGiud** [reply]
Well no you can't do that.

You reset MOOSE to this old version, you did not reset libmesh or timpi.
We do not advise you do that at all.
Please use the latest moose version and update your application to make it work with this latest MOOSE. This is the path of least effort and of least bugs!

**Reply by lindsayad** [reply]
Yea the rolled back moose is not going to work with the new submodule hashes. If you really want to use that moose version you should run the git submodule update command after checking out the old moose

**Reply by 09liu** [reply]
So I should use the follow command to aligne the version of moose and libmesh .
`cd ~/projects`
`git clone https://github.com/idaholab/moose.git`
`cd moose`
`git reset --hard ccf25ce`
`git checkout master`
`git submodule update --init`
`git submodule foreach --recursive git submodule update --init`

1. moose
commit ccf25cee294df35a2796b76e2df27558dac45122 (HEAD -> master, next)
Merge: 27d88630d4 3bb5876c5c
Author: moosetest <bounces@inl.gov>
Date:   Wed Jul 13 22:46:33 2022 -0600

    Merge commit '3bb5876c5c3f29649f426ff5535f8a3007658acb'

2.  libmesh
commit 4747096de5d6c69ed79f28e38ce45c76546364c3 (HEAD)
Merge: b50932009 fd6ba0e0a
Author: Roy H. Stogner <Roy.Stogner@inl.gov>
Date:   Fri Jun 3 13:23:45 2022 -0500

    Merge pull request #3311 from roystgnr/matrix_dual_decl
    
    Use declarations header for declarations

3. git submodule status:
e1211aaf2dec766e9a8e2da1631ee2f36ec7c5de large_media (remotes/origin/phase_field_old_wiki-67-ge1211aa)
 4747096de5d6c69ed79f28e38ce45c76546364c3 libmesh (cpp03_final-7575-g4747096de)
 f855b95493736b087b8ccc16dc6c5b29bc4b5aa8 petsc (v3.16.5-15-gf855b954937)

Now their versions seem to match, I will try to install again.
